Automattic/wp-calypso
# Limit Global Styles: Get correct blog ID for Atomic sites

#### Proposed Changes

The Global Styles gating logic performs a cutoff blog ID check to limit the feature on new sites only. However, it uses `get_current_blog_id` to determine the blog ID of the current site which doesn't return the WPCOM site ID for Atomic sites.

This PR updates the logic to use `Jetpack_Options::get_option( 'id' )` instead.

#### Testing Instructions

- Create a new WoA dev blog with a Business plan.
- Apply these changes to the Editing Toolkit plugin on the site.
- Additionally, apply these changes to manually remove the check that bails on Atomic sites:
```diff
diff --git a/apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php b/apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php
index c73b935d33..79e39bf2ae 100644
--- a/apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php
+++ b/apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/index.php
@@ -24,18 +24,7 @@ function wpcom_should_limit_global_styles( $blog_id = 0 ) {
 		return false;
 	}
 
-	if ( wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id ) ) {
-		return false;
-	}
-
-	// During the development stage, we only limit Global Styles on sites that have opted in via
-	// a blog sticker. This is a temporary check that will be removed as part of the public launch.
-	// Note that block stickers are not exposed by default to Atomic sites, so we're not limiting
-	// Global Styles on Atomic sites for now.
-	if ( ! function_exists( 'has_blog_sticker' ) ) {
-		return false;
-	}
-	return has_blog_sticker( 'wpcom-limit-global-styles', $blog_id );
+	return ! wpcom_site_has_feature( WPCOM_Features::GLOBAL_STYLES, $blog_id );
 }
 
 /**
```

- Go to SA and remove the Business plan.
- Go to the Site Editor and make sure the Global Styles are limited (upgrade modal, pre-save notice).
- Save some custom styles and visit the frontend.
- Make sure they are not visible.
